File: games/tictactoe/C/tictactoecmd.c
#include <stdio.h>
#include <stdbool.h>

void PrintBoard(char cells[])
{
  printf("\t Player (1) --> 'X' & Player (2) --> 'O'\n\n\n");
  printf("-------\n");
  printf("|%c|%c|%c|\n",cells[1],cells[2],cells[3]);
  printf("-------\n");
  printf("|%c|%c|%c|\n",cells[4],cells[5],cells[6]);
  printf("-------\n");
  printf("|%c|%c|%c|\n",cells[7],cells[8],cells[9]);
  printf("-------\n");
}

void InitializeBoard(char cells[],int size)
{
  for (int i = 0; i < size ; i++)
  {
    char let = i + 48;
    cells[i] = let;
  }
}

void UpdateBoard(char cells[],int *player_no, int *choice)
{
  if (*player_no == 1)
  {
    cells[*choice] = 'X';
  }
  if (*player_no == 2)
  {
    cells[*choice] = 'O';
  }
}

int IsWinner(char cells[], int *size)
{
  if (cells[1] == cells[2] && cells[2] == cells[3])
  {
    return 1;
  }
  else if (cells[4] == cells[5] && cells[5] == cells[6])
  {
    return 1;
  }
  else if (cells[7] == cells[8] && cells[8] == cells[9]) 
  {
    return 1;
  }
  else if (cells[1] == cells[4] && cells[4] == cells[7])
  {
    return 1;
  }
  else if (cells[2] == cells[5] && cells[5] == cells[8])
  {
    return 1;
  }
  else if (cells[3] == cells[6] && cells[6] == cells[9])
  {
    return 1;
  }
  else if (cells[1] == cells[5] && cells[5] == cells[9])
  {
    return 1;
  }
  else if (cells[3] == cells[5] && cells[5] == cells[7])
  {
    return 1;
  }
  else if (cells[1] != '1' && cells[2] != '2' && cells[3] != '3'
      && cells[4] != '4' && cells[5] != '5' && cells[6] != '6' && cells[7] != '7' && cells[8] != '8' && cells[9] != '9')
  {
    return 0;
  }
  else
  {
    return -1;
  }
}

int main()
{
  char cells[11];
  int player = 1;
  int choice;
  int i;
  
  int size = sizeof(cells)/sizeof(cells[0]); 
  InitializeBoard(cells,size);
  PrintBoard(cells);
  do {
    printf("Player %d, enter the square number: ",player);
    scanf("%d",&choice);
    UpdateBoard(cells, &player, &choice);
    PrintBoard(cells);
    i = IsWinner(cells,&size);
    if (player == 1)
    {
      player = 2;
    }
    else if (player == 2) {
      player = 1;
    }
  }while(i -= 1);
  
  if (i == 1)
  {
    printf("winner player %d\n",player);
  }
  else
  {
    printf("draw\n");
  }
  return 0;
}

File: games/tictactoe/python/tictactoe_cmd.py
import os
import itertools
#os.system("clear")

class Board:

    def __init__(self):
        self.cells = ["notaken"," "," "," "," "," "," "," "," "," "]

    def board_layout(self):
        print("-------")
        print("|%s|%s|%s|"%(self.cells[1],self.cells[2],self.cells[3]))
        print("-------")
        print("|%s|%s|%s|"%(self.cells[4],self.cells[5],self.cells[6]))
        print("-------")
        print("|%s|%s|%s|"%(self.cells[7],self.cells[8],self.cells[9]))
        print("-------")
    
    def update_board(self,player_no,player):
        if self.cells[player]==" ":
            if player_no == 1:
                self.cells[player] = "X"
            if player_no == 2:
                self.cells[player] = "O"
        
    def iswinner(self):
        setA = itertools.permutations([1,2,3])
        setB = itertools.permutations([4,5,6])
        setC = itertools.permutations([7,8,9])
        setD = itertools.permutations([1,5,9])
        setE = itertools.permutations([7,5,3])
        setF = itertools.permutations([1,4,7])
        setG = itertools.permutations([2,5,8])
        setH = itertools.permutations([3,6,9])
        
        for i in setA,setB,setC,setD,setE,setF,setG,setH:
            for k in list(i):
                if self.cells[k[0]] == self.cells[k[1]] == self.cells[k[2]] and self.cells[k[0]] != " ":
                    return True
    def isdraw(self):
        if " " in self.cells:
            return True
        return False

board = Board()
def refresh_screen(playerno,player):
    # refreshes our command line interface
    #os.system("clear")
    board.board_layout()
    board.update_board(playerno,player)
    board.iswinner()

board.board_layout()
while True:
    player1 = int(input("(X)>>INPUT A NUMBER BETWEEN 1-8: \n"))
    board.update_board(1,player1)
    board.board_layout()
    if board.isdraw() == False:
        print("Draw")
        break
    if board.iswinner() == True:
        print("Player 1 is winner")
        break

    player2 = int(input("(O)>>INPUT A NUMBER BETWEEN 1-8: \n"))
    board.update_board(2,player2)
    board.board_layout()
    if board.isdraw() == False:
        print("Draw")
        break
    if board.iswinner() == True:
        print("Player 2 is winner")
        break
     
